engine.py

```diff
@@ -108,6 +116,57 @@ def post_app_init(self):
             except:
                 traceback.print_exc()
 
+    def validate_context(self, tank, context):
```

Comment:
@kevinsallee After some thinking, I think you are right. I might have commented a bit too fast. The new core will probably remove the need for doing this. To make it clear, we could probably explain the reason of this method in the docstr, so everybody is on the same line  and we all know this part should go away once we upgrade to the latest Tank core version.

And just as a note, @frenchdog `engine_init` is not legacy code with our current Tank core version.